### PR TITLE
Select list width

### LINKED ISSUE #6861: Allow the width of a select dropdown (e.g. Filterable) to be wider than the original component.
### Description

When implementing a select component, in this case Filterable - the length of the text for the options in the dropdown often far exceed the width of the control itself. We need to be able to set the dropdown width to be larger than that of the main select component. 

### Suggested solution

Allow the dropdown width to be configurable and unlock this so that the width can exceed that of the parent control. 

### Demo URL

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by edleeks87**
fix #6861

### Proposed behaviour

Removes `left` and `right` options from `listPlacement` type, only `top` and `bottom` remain supported

BREAKING CHANGE: `listPlacement` prop only supports `top` and `bottom` now

Adds support for overriding the list width via `listWidth` prop. Adds support for placing
list at `top-start`, `top-end`, `bottom-start` and `botton-end`. When `top` or `bottom`
is passed to `listPlacement` it will internally append `-end` to it.

image

image

### Current behaviour

`listPlacement` supports `left` and `right`
no support for `listWidth`

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/select/simple-select/simple-select.pw.tsx
Comment (Parsium) [inline src/components/select/simple-select/simple-select.pw.tsx:1024]: **suggestion(non-blocking)**: what do you think about defining this value as a parameter for the tests? This would avoid the need for conditional logic within the test:

```tsx
([
  ["top", "top-end", "100px 300px", 150, 2],
  ["bottom", "bottom-end", "0px 300px", 150, 68],
] as const).forEach(([position, expectedPosition, margin, left, top] => {
   // test code...
});
```

File: src/components/select/simple-select/simple-select.pw.tsx
Comment (Parsium) [inline src/components/select/simple-select/simple-select.pw.tsx:1051]: **suggestion(non-blocking)**: another option we could consider here is using the `toBeCloseTo` assertion, which is designed for handling this kind of comparison. Not essential though - just something that might be helpful:

```suggestion
      expect(listLeftPosition).toBeCloseTo(left);
      expect(listTopPosition).toBeCloseTo(top);
      expect(listLeftPosition).toBeCloseTo(left);
      expect(listTopPosition).toBeCloseTo(top);
```

**Review by Parsium — APPROVED** [review]

File: src/components/select/simple-select/simple-select.pw.tsx
Comment (edleeks87) [reply src/components/select/simple-select/simple-select.pw.tsx:1024]: I think I'll leave this for now as there is other examples of the same approach in this file so I'd prefer it be tidied up by the code owners when they come to review the suite as a whole etc

File: src/components/select/simple-select/simple-select.pw.tsx
Comment (edleeks87) [reply src/components/select/simple-select/simple-select.pw.tsx:1051]: Good point I'll update this before I merge

**Review by Parsium — DISMISSED** [review]

**Review by Parsium — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by dariusbercea-sage — APPROVED** [review]

**Pull request merged**
